runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -692,9 +692,40 @@ TR_J9ServerVM::getStaticReferenceFieldAtAddress(uintptrj_t fieldAddress)
 bool
 TR_J9ServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *clazz)
    {
+   if(!method)
+      {
+      return false;
+      }
+
    JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, method, clazz);
-   return std::get<0>(stream->read<bool>());
+   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
+   
+   if (vmInfo->_jlrMethodInvoke == NULL)
+      {
+      stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, JITServer::Void());
+      vmInfo->_jlrMethodInvoke = std::get<0>(stream->read<J9Method*>());
+      }
+   // If _jlrMethodInvoke is NULL at the client, return true
+   if (vmInfo->_jlrMethodInvoke == ((J9Method *) method) || vmInfo->_jlrMethodInvoke == NULL)
```

Comment:
Ideally the test `vmInfo->_jlrMethodInvoke == NULL` is performed before `vmInfo->_jlrMethodInvoke == ((J9Method *) method)`. Lets push that `vmInfo->_jlrMethodInvoke == NULL` test above like this:
```
if (vmInfo->_jlrMethodInvoke == NULL)
      {
      // Cached information could be outdated. Ask the client again.
      stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, JITServer::Void());
      vmInfo->_jlrMethodInvoke = std::get<0>(stream->read<J9Method*>());
      if (vmInfo->_jlrMethodInvoke == NULL)
         return true;
      }
if (vmInfo->_jlrMethodInvoke == ((J9Method *) method)
    return true;
```